### DISCUSSION TITLE
mTLS support for authentication in Kafka integration

### DISCUSSION THREAD

**Discussion by radic** [milvus-io/milvus#27783] [Ideas & Feature requests]
I would love to raise a feature request about mTLS support for authentication in Kafka integration.

Kafka supports mTLS: [Encrypt and Authenticate with TLS](https://docs.confluent.io/platform/current/kafka/authentication_ssl.html), which is more secure. It will be great if we can integrate our Kafka with Milvus in this approach.

Since certificate rotation will be enabled for better security, it is expect that Milvus can also periodically refresh cert before the cert gets expired.

**Comment by yhmo** [comment]
Similar request:
https://github.com/milvus-io/milvus/discussions/27486

A feature request is created here: https://github.com/milvus-io/milvus/issues/27563
